q.js

```diff
@@ -1580,6 +1580,7 @@ function all(promises) {
                     promise,
                     function (value) {
                         promises[index] = value;
+                        deferred.notify( { index: index, pending: pendingCount - 1 } );
```

Comment:
Note that there is existing code, below, that propagates a different kind of progress notification.